Readability.js

```diff
@@ -961,6 +961,18 @@ Readability.prototype = {
           return null;
         }
       } else {
+        // Find out text direction from ancestors of final top candidate.
+        var ancestors = [parentOfTopCandidate].concat(this._getNodeAncestors(parentOfTopCandidate));
```

Comment:
We only check the things explicitly in the array, right? So if we just concat `[topCandidate, parentOfTopCandidate]` with the ancestors of the `parentOfTopCandidate`, I don't think we check anything else?

While it's true that we would do the correct things in terms of display of the article itself, that wouldn't be the case for the article title and byline (which are likely also rtl in cases like this) because the metadata on our readability detection object would still be wrong.